### DISCUSSION TITLE
Looking for a few clarifications regarding authors' roles

### DISCUSSION THREAD

**Discussion by arnaudgallou** [quarto-dev/quarto-cli#10663] [Q&A] [answered]
### Description

I'm making some changes to one of my package that generates Quarto's author/affiliations schemas from tabular data. Currently, I use an array structure to list roles, whether they're roles from the Contributor Roles Taxonomy or custom roles. Looking at the ['Author Roles' section](https://quarto.org/docs/authoring/front-matter.html#roles) again, the text and examples make it sound like that the array structure is reserved to CRediT roles, when custom roles should be written as free form text. Is this correct? Or can one use the array structure to list cutom roles too. E.g.:

```
roles:
  - Conceived the study
  - Analysed the results
  - Wrote the manuscript
```

Also, having a quick look at the lua filter that handles roles, it seems that CRediT roles have to be in lower case. Am I right?

**Comment by mcanouil** [comment, marked as answer]
See <https://quarto.org/docs/journals/authors.html#roles> and the whole page for detailed documentation on the authors schema.

> Also, having a quick look at the lua filter that handles roles, it seems that CRediT roles have to be in lower case. Am I right?

Please link to code when referring to some part of the codebase.

**Reply by arnaudgallou** [reply]
Thanks, it's clearer on the page you linked. It's a bit weird to have the same type of information spread over different pages though. Why not merging the content of the page you linked with the front matter one?

Here're the links to the specific code parts:
https://github.com/quarto-dev/quarto-cli/blob/f65180a75e1cf2996328cd51cb4fd5d02d391511/src/resources/filters/modules/authors.lua#L120-L144C2

https://github.com/quarto-dev/quarto-cli/blob/f65180a75e1cf2996328cd51cb4fd5d02d391511/src/resources/filters/modules/authors.lua#L646-L667C4

I'm realizing I missed the `local term = string.lower(role.role)` line the first time. So it's case insensitive. Makes more sense.

**Reply by mcanouil** [reply]
I suggest you open a documentation issue using the GitHub issue template about this. Thank you!
